// 首行:语法指定行
syntax="proto3";
package contacts;

import "google/protobuf/any.proto";

//定义联系人message

message Phone{
    string phone_number = 1;
    enum PhoneType{
        MP = 0; // 移动电话
        TEL = 1; // 固定电话
    }
    PhoneType type = 2;
}

message Address{
    string home_addr = 1; // 家庭地址
    string unit_addr = 2; // 单位地址
}

message PeopleInfo{
    string name = 1;
    int32 age = 2;
    repeated Phone phone = 3;
    google.protobuf.Any addr = 4;
    oneof other_contact{
        string qq = 5;
        string wechat = 6;
    }
    map<string,string> remark = 7;
}

message Contacts{
    repeated PeopleInfo contacts = 1;
}